Upstream commit: acdc7ce30e58088083421fa395dec8235c635635
ChangeLog / commit message:
* config/obj-elf.c (obj_elf_section_change_hook): New function.
	* config/obj-elf.h (obj_elf_section_change_hook): Declare it.
	* config/tc-mips.c (s_change_sec): Call it if OBJ_ELF.
Files: gas/config/tc-mips.c

--- a/gas/config/tc-mips.c
+++ b/gas/config/tc-mips.c
@@ -9031,16 +9031,6 @@
       && (sec == 'd' || sec == 'r'))
     sec = 's';
 
-#ifdef OBJ_ELF
-  /* The ELF backend needs to know that we are changing sections, so
-     that .previous works correctly.  We could do something like check
-     for a obj_section_change_hook macro, but that might be confusing
-     as it would not be appropriate to use it in the section changing
-     functions in read.c, since obj-elf.c intercepts those.  FIXME:
-     This should be cleaner, somehow.  */
-  obj_elf_section_change_hook ();
-#endif
-
   mips_emit_delays (false);
   switch (sec)
     {
